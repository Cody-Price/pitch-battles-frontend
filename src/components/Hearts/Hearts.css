.hearts {
  position: absolute;
}

.heart {
  width: 15px;
  height: 15px;
  background: url("./heart.png");
  background-size: cover;
  background-repeat: no-repeat;
  margin-bottom: 2px;
}

.hearts.player-hearts {
  left: -10px;
}

.hearts.monster-hearts {
  right: -10px;
}
