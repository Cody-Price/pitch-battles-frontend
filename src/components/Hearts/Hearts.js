import React, { Component } from "react";
import "./Hearts.css";

class Hearts extends Component {
  render() {
    const hearts = this.props.hearts.map(heart => {
      return <div className="heart" />;
    });
    return (
      <section className={`hearts ${this.props.character}`}>{hearts}</section>
    );
  }
}

export default Hearts;
