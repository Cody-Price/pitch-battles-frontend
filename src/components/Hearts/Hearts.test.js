import React from "react";
import ReactDOM from "react-dom";
import Hearts from "./Hearts";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Hearts", () => {
  let wrapper;
  const mockHearts = [{ position: "hello" }];

  beforeEach(() => {
    wrapper = shallow(<Hearts hearts={mockHearts} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
