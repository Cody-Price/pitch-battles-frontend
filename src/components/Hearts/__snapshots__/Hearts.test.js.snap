// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hearts should match the snapshot 1`] = `
<section
  className="hearts undefined"
>
  <div
    className="heart"
    key="hello"
  />
</section>
`;
